src/EventSubscriber/CacheSubscriber.php

```diff
@@ -237,7 +238,9 @@ public function onResponse(FilterResponseEvent $event) {
     $expire = $this->maxAgeToExpire($metadata->getCacheMaxAge());
 
     // Write the cache entry for the cache metadata.
-    $ccid = $this->getCacheIdentifier($this->metadata);
+
+    $queryHash = md5(serialize($request->attributes->get('queries')));
```

Comment:
This should be a cache context instead, no? That's the whole point of cache contexts ;)